src/calendar-api.ts

```diff
@@ -0,0 +1,56 @@
+//function to fetch events from cloud API endpoint
+
+export const fetchGoogleCalendarEvents = async () => {
+	const apiKey = import.meta.env.VITE_AWS_API_KEY; //secured API key
+	const apiUrl = "https://ghvy6isiq6.execute-api.eu-north-1.amazonaws.com/dev/events"; //AWS API url
+
+	interface CalendarEvent {
+		start?: {
+			dateTime?: string; // ISO for events with specific times
+			date?: string; // ISO for all-day events
+		};
+		end?: {
+			dateTime?: string; // ISO for events with specific times
+			date?: string; // ISO for all-day events
+		};
+	}
+
+	try {
+		const response = await fetch(apiUrl, { method: "GET", headers: { "x-api-key": apiKey } });
+		const data = await response.json();
+		const events: CalendarEvent[] = JSON.parse(data.body);
+
+		//getting an array of days for events
+		const getAllDaysInRange = (startDate: string, endDate: string): string[] => {
+			const start = new Date(startDate);
+			const end = new Date(endDate);
+			const dates: string[] = [];
+
+			for (let date: Date = new Date(start); date <= end; date.setDate(date.getDate() + 1)) {
```

Comment:
```suggestion
			for (let date: Date = start; date <= end; date.setDate(date.getDate() + 1)) {
```

I don't think you need to convert this into a Date again, since you've done that already on line 25.